agent/agents/mysql/slowlog/parser/parser.go

```diff
@@ -167,90 +168,116 @@ func (p *SlowLogParser) Run() {
 func (p *SlowLogParser) parseHeader(line string) {
 	p.logf("header")
 
-	if !headerRe.MatchString(line) {
+	if !isHeader(line) {
 		p.inHeader = false
 		p.inQuery = true
 		p.parseQuery(line)
 		return
 	}
+	line = skipPrefix(line)
 
 	if p.headerLines == 0 {
 		p.event.Offset = p.lineOffset
 	}
 	p.headerLines++
 
 	switch {
-	case strings.HasPrefix(line, "# Time"):
-		p.logf("time")
-		m := timeRe.FindStringSubmatch(line)
+	case strings.HasPrefix(line, "Time: "):
+		p.parseTime(line)
+
+	case strings.HasPrefix(line, "User"):
+		p.parseUser(line)
+
+	case strings.HasPrefix(line, "admin"):
+		p.parseAdmin(line)
+
+	default:
+		p.parseMetrics(line)
+	}
+}
+
+func skipPrefix(line string) string {
+	line = line[2:]
+	return line
+}
+
+func (p *SlowLogParser) parseTime(line string) {
+	p.logf("time")
+	m := timeRe.FindStringSubmatch(line)
+	if len(m) == 2 {
+		p.event.Ts, _ = time.ParseInLocation("060102 15:04:05", m[1], p.opts.DefaultLocation)
+	} else {
+		m = timeNewRe.FindStringSubmatch(line)
 		if len(m) == 2 {
-			p.event.Ts, _ = time.ParseInLocation("060102 15:04:05", m[1], p.opts.DefaultLocation)
+			p.event.Ts, _ = time.ParseInLocation(time.RFC3339Nano, m[1], p.opts.DefaultLocation)
 		} else {
-			m = timeNewRe.FindStringSubmatch(line)
-			if len(m) == 2 {
-				p.event.Ts, _ = time.ParseInLocation(time.RFC3339Nano, m[1], p.opts.DefaultLocation)
-			} else {
-				return
-			}
-		}
-		if userRe.MatchString(line) {
-			p.logf("user (bad format)")
-			m := userRe.FindStringSubmatch(line)
-			p.event.User = m[1]
-			p.event.Host = m[2]
-		}
-
-	case strings.HasPrefix(line, "# User"):
-		p.logf("user")
-		m := userRe.FindStringSubmatch(line)
-		if len(m) < 3 {
 			return
 		}
+	}
+	if userRe.MatchString(line) {
+		p.logf("user (bad format)")
+		m := userRe.FindStringSubmatch(line)
 		p.event.User = m[1]
 		p.event.Host = m[2]
+	}
+}
 
-	case strings.HasPrefix(line, "# admin"):
-		p.parseAdmin(line)
+func (p *SlowLogParser) parseUser(line string) {
+	p.logf("user")
+	m := userRe.FindStringSubmatch(line)
+	if len(m) < 3 {
+		return
+	}
+	p.event.User = m[1]
+	p.event.Host = m[2]
+}
 
-	default:
-		p.logf("metrics")
-		submatch := schema.FindStringSubmatch(line)
-		if len(submatch) == 2 {
-			p.event.Db = submatch[1]
+func (p *SlowLogParser) parseMetrics(line string) {
+	p.logf("metrics")
+	submatch := schema.FindStringSubmatch(line)
+	if len(submatch) == 2 {
+		p.event.Db = submatch[1]
+	}
+
+	if !strings.Contains(line, ":") {
+		return
```

Comment:
Same question as above: do we need to log line which cannot be parsed?

Reply:
added logging for this case